### PR TITLE
Helm-415: Docs direct user to create a Custom Action when creating a Dashboard but this feature no longer exists in Grafana v9

### PR DESCRIPTION
**Pull request by Bonrob2**
# Pull Request Check Sheet

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://opennms.atlassian.net/projects/OPG)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new feature, is there documentation?
* [ ] If this is a new feature or substantial change, are there tests that cover it?

# Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

Once the reviewer(s) accept the PR and the branch passes continuous integration in Bamboo, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

# External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/HELM-415
* Continuous Integration: [CircleCI](https://circleci.com/gh/OpenNMS/grafana-plugin)

### CODE REVIEW

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 1]
```diff
@@ -186,3 +153,18 @@ For nodes, you can configure only a table graph.
 ** (Optional) Configure the settings in the bottom half of the panel to customize the graph.
 . Update the panel's *Name*.
 . Click *Save* at the top-right of the panel.
+
+== Use data links to provide alarm context
+
+Data links in Grafana let you create a user-defined operation (for example, a link to a node detail page or a knowledge base article, links based on alarm severity, and so on) that might help users diagnose an alarm.
+Once created, a link to the specified data will appear when you right-click an alarm in an alarms table:
+
+image::gf-data-link.png["Alarms table with three alarms. The context menu is displayed, listing three defined options."]
```
Comment (synqotik) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:162]: This is an image of the Alarm Table OpenNMS context menu, but not the Grafana Data Link context menu.

**Review by synqotik — CHANGES_REQUESTED** [review]
Looks good, except see my comment, `gf-data-link.png` needs to be updated.

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 2]
```diff
@@ -77,10 +45,7 @@ See the https://grafana.com/docs/grafana/latest/datasources/[official Grafana do
...
 
 . Navigate to the home page of your Grafana instance.
-. In the left navigation menu, select menu:Configuration[Data Sources]:
-+
-image::gf-data-sources.png["Grafana menu displaying navigation options. The Data Sources option is highlighted.", 250]
-
+. In the left navigation menu, select menu:Administration[Data ources].
 . Select *Add Data Source*.
 . In the *Search* box, type *OpenNMS Entities*.
 . Select the *OpenNMS Entities* datasource.
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:51]: User will want to repeat the same steps with a Performance datasource, and optionally a Flow datasource.
Personally, I would suggest stepping through a Perf DS first, as that would become the default datasource

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 3]
```diff
@@ -70,7 +70,7 @@ For information on displaying other data in a dashboard, see <<bw-panel-create,
 . In the left navigation menu, click *Dashboards*.
 . Click menu:New[New Dashboard].
 . Click *Add Visualization*.
-. Choose your data source (for example, *OpenNMS Entities*).
+. Choose your data source (for example, *OpenNMS Performance*).
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:73]: This must be Entities for an Alarm Table visualization.
```suggestion
. Choose your data source (for example, *OpenNMS Entities*).
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 4]
```diff
@@ -28,30 +28,30 @@ image::gf-plugins.png["Grafana menu displaying navigation options. The Plugins o
...
 [[bw-ds-setup]]
 == Create a datasource
 
 The type of data that you can display in a dashboard depends on the datasource that you select.
 
 Before you create a dashboard, you must set up the datasource (xref:datasources:performance_datasource.adoc[performance], xref:datasources:flow_datasource.adoc[flow], or xref:datasources:entity_datasource.adoc[entity]).
-For this walk-through, we will configure an entity datasource.
-Note that you can follow the same process to configure a flow or performance datasource.
+For this walk-through, we will configure a performance datasource.
+Note that you can follow the same process to configure an entity or flow datasource.
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:40]:
```suggestion
For this walk-through, we will configure a entity datasource.
After completing these steps, repeat the same process to configure a performance datasource, and optionally a flow datasource if you are collecting flow data.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 5]
```diff
@@ -28,30 +28,30 @@ image::gf-plugins.png["Grafana menu displaying navigation options. The Plugins o
...
 
 . Navigate to the home page of your Grafana instance.
-. In the left navigation menu, select menu:Administration[Data ources].
-. Select *Add Data Source*.
-. In the *Search* box, type *OpenNMS Entities*.
-. Select the *OpenNMS Entities* datasource.
+. In the left navigation menu, select menu:Administration[Data Sources].
+. Select *Add New Data Source*.
+. In the *Search* box, type *OpenNMS Performance*.
+. Select the *OpenNMS Performance* datasource.
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:51]:
```suggestion
. In the left navigation menu, select menu:Administration[Data Sources].
. Select *Add New Data Source*.
. In the *Search* box, type *OpenNMS Entities*.
. Select the *OpenNMS Entities* datasource.
```

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 6]
```diff
@@ -28,30 +28,30 @@ image::gf-plugins.png["Grafana menu displaying navigation options. The Plugins o
...
 The settings dialog is displayed:
 +
-image::gf-data-config.png["OpenNMS Entities datasource settings dialog displaying sample settings.", 500]
+image::gf-data-config2.png["OpenNMS Performance datasource settings dialog displaying sample settings.", 500]
```
Comment (mmahacek) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:54]:
```suggestion
image::gf-data-config.png["OpenNMS Entities datasource settings dialog displaying sample settings.", 500]
```

**Review by mmahacek — COMMENTED** [review]
Reverting some of my past suggestions.  An Entity Datasource is required for the following steps in the walkthrough.  I was just suggesting Perf first as it becomes the default datasource when building panels, and Perf is typically used more than Entity when building metric dashboards.

File: docs/modules/getting_started/pages/basic_walkthrough.adoc (hunk 3 above)
Comment (Bonrob2) [reply docs/modules/getting_started/pages/basic_walkthrough.adoc:73]: Thanks. This documentation was the first thing I wrote when I started at ONMS 4 years ago. I believe Jesse had suggested starting with Entities data source. We could create a ticket to change the walkthrough to be Performance data source focused if that makes more sense.

**Review by mmahacek — APPROVED** [review]

File: docs/modules/getting_started/pages/basic_walkthrough.adoc [hunk 7]
```diff
@@ -186,3 +153,16 @@ For nodes, you can configure only a table graph.
 ** (Optional) Configure the settings in the bottom half of the panel to customize the graph.
 . Update the panel's *Name*.
 . Click *Save* at the top-right of the panel.
+
+== Use data links to provide alarm context
+
+Data links in Grafana let you create a user-defined operation (for example, a link to a node detail page or a knowledge base article, links based on alarm severity, and so on) that might help users diagnose an alarm.
+Once created, a link to the specified data will appear when you right-click an alarm in an alarms.
```
Comment (synqotik) [inline docs/modules/getting_started/pages/basic_walkthrough.adoc:160]: Change to something like:

Once created, links will appear in the Alarm Table data which can be left-clicked. If you have configured more than one Data Link for the panel, left-clicking an item will cause a context menu to display which allows you to choose which link to follow.

**Pull request merged**